node/node.go

```diff
@@ -194,22 +195,24 @@ func activateShhService(stack *node.Node, config *params.NodeConfig, db *leveldb
 		logger.Info("SHH protocol is disabled")
 		return nil
 	}
-	if err := stack.Register(func(*node.ServiceContext) (node.Service, error) {
-		return timesource.Default(), nil
-	}); err != nil {
-		return err
+	if config.WhisperConfig.EnableNTPSync {
+		if err := stack.Register(func(*node.ServiceContext) (node.Service, error) {
+			return timesource.Default(), nil
+		}); err != nil {
+			return err
+		}
 	}
 
 	err = stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
-		var timeSource *timesource.NTPTimeSource
-		if err := ctx.Service(&timeSource); err != nil {
-			return nil, err
-		}
 		whisperServiceConfig := &whisper.Config{
 			MaxMessageSize:     whisper.DefaultMaxMessageSize,
 			MinimumAcceptedPOW: 0.001,
-			TimeSource:         timeSource.Now,
 		}
+
+		if whisperServiceConfig.TimeSource, err = whisperTimeSource(ctx, config.WhisperConfig.EnableNTPSync); err != nil {
```

Comment:
i would do this condition in place, as it doesn't add much to readability

Reply:
cheers, it looked like a good option on a first instance but its too large to make it readable.